src/main/java/org/mule/extension/file/common/api/FileSystem.java

```diff
@@ -79,6 +79,31 @@ default List<Result<InputStream, A>> list(FileConnectorConfig config,
     return list(config, directoryPath, recursive, matcher);
   }
 
+  /**
+   * Lists and read all the files in the {@code directoryPath} which match the given {@code matcher}.
+   * <p>
+   * If the listing encounters a directory, the output list will include its contents depending on the value of the
+   * {@code recursive} argument. If {@code recursive} is enabled, then all the files in that directory will be listed immediately
+   * after their parent directory.
+   * <p>
+   *
+   * @param config                the config that is parameterizing this operation
+   * @param directoryPath         the path to the directory to be listed
+   * @param recursive             whether to include the contents of sub-directories
+   * @param matcher               a {@link Predicate} of {@link FileAttributes} used to filter the output list
+   * @param timeBetweenSizeCheck  wait time between size checks to determine if a file is ready to be read in milliseconds.
+   * @return a {@link List} of {@link Result} objects, each one containing each file's content in the payload and metadata in the
+   *         attributes
+   * @throws IllegalArgumentException if {@code directoryPath} points to a file which doesn't exist or is not a directory
+   */
+  default List<Result<InputStream, A>> listAndRead(FileConnectorConfig config,
```

Comment:
We need to test this new functionality

Reply:
Yes but this is just the interface we will test it in the file connector